msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/MsalAzureSDKException.java

```diff
@@ -0,0 +1,14 @@
+package com.microsoft.aad.msal4j;
+
+/**
+ * Exception type thrown when Azure SDK returns an error response.
+ */
+public class MsalAzureSDKException extends MsalException{
```

Comment:
This whole AppTokenProvider flow is intended for the Azure SDK, but in theory anyone who wants to implement their own auth logic could use it, right? 

If so, then maybe this should just be something like 'TokenProviderException'. On the other hand, maybe it's better to leave this specific to Azure SDK to discourage other people from trying to use it.

Reply:
That's a good point. This functionality is meant to be used by the Azure SDK team only. I have added documentation to discourage any other teams from using it.